config/initializers/lti_authentication.rb

```
@@ -0,0 +1,55 @@
+require 'lti_launch'
+
+module LtiAuthentication
+
+  LTI_AUTH_HEADER_PREFIX='LtiState'.freeze
+
+  class WardenStrategy < Warden::Strategies::Base
+    
+    # True if this strategy should be run for this request
+    def valid?
+      @lti_state = fetch_state
+      @lti_state.present?
+    end
+
+    def authenticate!
```

Nit: can we structure this to fail fast and error out? E.g.: 

```
unless ll = LtiLaunch.is_valid?(@lti_state)
  Rails.logger.warn("LtiAuthentication::WardenStrategy couldn't find LtiLaunch with state = '#{@lti_state}'")
  return handle_failure(:forbidden)
end

unless user = ll.user
  Rails.logger.warn("LtiAuthentication::WardenStrategy couldn't find user with canvas_id = #{ll.request_message.canvas_user_id}")
  return handle_failure(:forbidden)
end

Rails.logger.debug("Authenticated user_id = #{user.id} using LtiAuthentication::WardenStrategy")
success(!user)
```